src/toolbar_gui.cpp

```diff
@@ -930,7 +930,11 @@ static CallBackFunction MenuClickBuildRoad(int index)
 
 static CallBackFunction ToolbarBuildWaterClick(Window *w)
 {
-	PopupMainToolbMenu(w, WID_TN_WATER, STR_WATERWAYS_MENU_WATERWAYS_CONSTRUCTION, 1);
+	DropDownList *list = new DropDownList();
+
+	list->push_back(new DropDownListIconItem(SPR_IMG_BUILD_CANAL, PAL_NONE, STR_WATERWAYS_MENU_WATERWAYS_CONSTRUCTION, WATER_CLASS_CANAL, false));
+
+	ShowDropDownList(w, list, 1, WID_TN_WATER, 140, true, true);
```

Comment:
Just use `0` as the value and default value here (and below) as there is only option and the value is not checked. Definitely wrong to use in the list item `WATER_CLASS_CANAL` and `1` as the default value.

Reply:
Which list item should I use instead?
Also, using 0 as the default value doesn't work in neither Waterways nor Airports. I have to use `1` or `2` for it to work. Or is that related to the class?

Reply:
Set the value to 0 as well. No reason to use either of those enum values.